### DISCUSSION TITLE
Can't get past need for IIB_SECRET_KEY despite proper configuration

### DISCUSSION THREAD

**Discussion by coderCK2** [zanllp/infinite-image-browsing#491] [Q&A] [answered]
Hello all, I'm having an issue where upon installing the extension in A1111 for the first time, I am presented with the modal/dialog box stating:
> Secret Key must be configured
> For security reasons, you must separately configure Secret Key for this extension, refer to the IIB_SECRET_KEY in the .env.example file under the root directory of this extension. 
>   This error only appears when gradio-auth is configured.

I absolutely have copied the `.env.example` file as `.env`, and have chosen a string for the IIB_SECRET_KEY, which I saved into the `.env` file like `IIB_SECRET_KEY=secretkey`. It didn't work right away, so I did a Reload UI but was still presented with the dialog. I then did a complete exit and restart of the web UI process on my machine (Win10), but was still presented with the dialog. I decided to then remove the secret key and instead set `IIB_ACCESS_CONTROL=disable`, but I am _still_ presented with the dialog. Any ideas what could be going wrong? I have a feeling it's something simple but I'm just missing it.

EDIT: I also just tried adding `set IIB_ACCESS_CONTROL=disable` to `webui-user.bat` (based on what I found in [this post](https://github.com/zanllp/sd-webui-infinite-image-browsing/discussions/236#discussioncomment-6198082)) and restarting, but it also did not work.

**Comment by coderCK2** [comment]
I've tried all kinds of things to get this working, even changing the line endings of `.env` from Windows to Unix. In the end, I redefined the function `get_secret_key_required()` (in `scripts\iib\tool.py`) as `return False`. That finally did the trick.

FWIW, I think that the line `secret_key = os.getenv("IIB_SECRET_KEY")` in `api.py` is failing to retrieve whatever value I set in `.env`, because if I hardcode it to be `secret_key = "test"` (for example), it works.

**Comment by zanllp** [comment]
Did you place your .env file correctly?

<img width="302" alt="image" src="https://github.com/zanllp/sd-webui-infinite-image-browsing/assets/25872019/6b624165-a10b-41bd-a803-0219957a827d">

**Comment by coderCK2** [comment]
> Did you place your .env file correctly?

Yes, I copied the file in place, renamed it, and modified the copy.
![image](https://github.com/zanllp/sd-webui-infinite-image-browsing/assets/42456814/6d58bb5a-2530-47d1-be42-73ee573f4d99)

**Comment by zanllp** [comment]
I cannot reproduce this issue. I tried two different methods, and both of them worked fine. This limitation has been in place for several months, and there haven't been any similar reports before. It's possible that there is an exception occurring with the dotenv package.

**Reply by coderCK2** [reply]
I had an idea that maybe this was happening because I was launching `webui-user.bat` from a Git bash for Windows prompt, but even when running from a regular Windows command prompt I am having this issue. So strange.

**Comment by sajb0t** [comment]
Same problem here. Had to stop using this extension and change to other image browser :/

**Reply by coderCK2** [reply]
Did you see my [comment here](https://github.com/zanllp/sd-webui-infinite-image-browsing/discussions/491#discussioncomment-8019949)? If you make the same change that I describe, it will work. I've tried other image browsers, and despite this issue, it's the best.

**Comment by coderCK2** [comment, marked as answer]
@zanllp I figured it out just now. After updating the extension and having to "fix" this once again, I spent some time getting to the bottom of it. It turns out that if I start A1111 from an elevated prompt, the issue is fixed.

It was my comment [here](https://github.com/zanllp/sd-webui-infinite-image-browsing/discussions/491#discussioncomment-8019949) that led me down the right path to solving the issue, as it didn't seem right that a call to `os.getenv()` was failing, unless there was a permissions issue. Running A1111 as administrator allowed that call to function correctly on my system, resolving the problem.

I don't know if every other person using the extension is running as admin, or if there's just something configured strangely with my system, but it might be worth noting somewhere that in order for authentication/secret keys to work correctly in Windows, the user *may* have to run as administrator.
